packages/p2p-v2/src/components/TextArea/TextArea.tsx

```diff
@@ -1,31 +1,63 @@
-import React, { useState } from 'react';
+import React, { HtmlHTMLAttributes, useState } from 'react';
+import clsx from 'clsx';
+import { Text } from '@deriv-com/ui/dist/components/Text';
 import './TextArea.scss';
 
-type TTextAreaProps = {
+type TTextAreaProps = HtmlHTMLAttributes<HTMLTextAreaElement> & {
+    hint?: string;
+    isInvalid?: boolean;
     label?: string;
+    maxLength?: number;
     onChange: React.ChangeEventHandler<HTMLTextAreaElement>;
-    placeholder: string;
+    placeholder?: string;
     shouldShowCounter?: boolean;
     value?: string;
 };
-const TextArea = ({ label, onChange, placeholder, shouldShowCounter = true, value }: TTextAreaProps) => {
+const TextArea = ({
+    hint,
+    isInvalid = false,
+    label,
+    maxLength,
+    onChange,
+    placeholder,
+    shouldShowCounter = false,
+    value,
+}: TTextAreaProps) => {
     const [currentValue, setCurrentValue] = useState(value);
 
     return (
-        <div className='p2p-v2-textarea'>
+        <div
+            className={clsx('p2p-v2-textarea', {
+                'p2p-v2-textarea--error': isInvalid,
+            })}
+        >
             <textarea
+                data-has-value={!!currentValue}
                 onChange={event => {
                     setCurrentValue(event.target.value);
                     onChange?.(event);
                 }}
                 placeholder={placeholder}
-            >
-                {value}
-            </textarea>
-
+                value={currentValue}
+            />
+            {label && (
+                <label>
+                    <Text color={isInvalid ? 'error' : 'less-prominent'} size='sm'>
+                        {label}
+                    </Text>
+                </label>
+            )}
             <div className='p2p-v2-textarea__footer'>
-                {label && <label>This information will be visible to everyone.</label>}
-                {shouldShowCounter && <span>{currentValue?.length || 0}/300</span>}
+                {hint && (
+                    <Text as='p' color={isInvalid ? 'error' : 'less-prominent'} size='xs'>
+                        {hint}
+                    </Text>
+                )}
+                {shouldShowCounter && (
+                    <Text color={isInvalid ? 'error' : 'less-prominent'} size='xs'>
+                        {currentValue?.length || 0}/{maxLength}
+                    </Text>
+                )}
```

Comment:
i think this might have conflicts with @nada-deriv's PR, she separated the hint out 👀 better discuss it with her, but this way is better IMO 👍